internal/mode/static/nginx/config/servers.go

```diff
@@ -487,6 +493,50 @@ func generateProxySetHeaders(filters *dataplane.HTTPFilters) []http.Header {
 	return append(proxySetHeaders, headers...)
 }
 
+func generateAddResponseHeaders(filters *dataplane.HTTPFilters) []http.Header {
+	headers := make([]http.Header, len(baseHeaders))
+	copy(headers, baseHeaders)
+
+	if filters == nil || filters.ResponseHeaderModifiers == nil {
+		return headers
+	}
+	headerFilter := filters.ResponseHeaderModifiers
+
+	headerLen := len(headerFilter.Add) + len(headers)
+	responseAddHeaders := make([]http.Header, 0, headerLen)
+	if len(headerFilter.Add) > 0 {
+		// FIXME(kevin85421): Should we use a different function?
+		addHeaders := convertSetHeaders(headerFilter.Add)
```

Comment:
Should we rename the function or create a new function?

Reply:
Let's rename. 

`convertSetHeaders` is just converting  `[]dataplane.HTTPHeader` to `[]http.Header`.  Let's name it something that describes that behavior.

I think `convertAddHeaders`  can be renamed as well. It does a similar thing as `convertSetHeaders` except it generates a map variable name.

Reply:
Do you have any preferred function name?

Reply:
Naming is hard 😅 Maybe `createHeaders` and `createHeadersWithVarName`? Open to other names though

Reply:
Update https://github.com/nginxinc/nginx-gateway-fabric/pull/1494/commits/d316a416f50524fd4de0dbb318d3c864e543b835 https://github.com/nginxinc/nginx-gateway-fabric/pull/1494/commits/9bf3447aecf8594e359ae07a61971e09adf64323